src/main/java/com/hms_networks/sc/canary/api/CanaryApiRequestSender.java

```diff
@@ -169,18 +174,85 @@ private static CanaryApiResponseStatus checkGenericResponseJson(
 
     // Check for "Errors" field
     if (response.has(ERROR_JSON_FIELD_NAME)) {
+      // Get the errors array
       JSONArray responseErrors = response.getJSONArray(ERROR_JSON_FIELD_NAME);
+
+      // Log errors from responseErrors. Duplicate messages should be counted and shown once.
       if (responseErrors.length() > 0) {
-        status = CanaryApiResponseStatus.UNKNOWN_ERROR;
-        Logger.LOG_INFO("There were " + responseErrors.length() + " errors found.");
-        for (int i = 0; i < responseErrors.length(); i++) {
-          Logger.LOG_CRITICAL("API error: " + responseErrors.getString(i));
+        // Log the number of errors in the response
+        Logger.LOG_INFO(
+            "The API response contained " + responseErrors.length() + " error message(s).");
+
+        // Get the first error message
+        final int firstErrorIndex = 0;
+        String apiErrorMessage = responseErrors.getString(firstErrorIndex);
+        int apiErrorMessageCount = 1;
+
+        // Loop through the rest of the error messages (if any)
+        for (int apiErrorMessageIndex = 1;
+            apiErrorMessageIndex < responseErrors.length();
+            apiErrorMessageIndex++) {
+          String nextApiErrorMessage = responseErrors.getString(apiErrorMessageIndex);
+          if (nextApiErrorMessage.equals(apiErrorMessage)) {
+            apiErrorMessageCount++;
+          } else {
+            // Check if error message suggests data duplication
+            if (checkApiErrorSuggestsDataDuplication(apiErrorMessage)) {
+              status = CanaryApiResponseStatus.ERROR_WAIT_FOR_EXPIRE;
+            }
+
+            // Log the error message
+            logApiError(apiErrorMessage, apiErrorMessageCount);
+
+            // Update the current error message and count
+            apiErrorMessage = nextApiErrorMessage;
+            apiErrorMessageCount = 1;
```

Comment:
This is a bit confusing to me and suggests that you might need more commenting. IF the error message is not an apierrormessage why does the count get reset to 1?

Reply:
This logic counts identical error messages, and in this particular condition, the count is reset to 1 because the current message does/did not equal the next message.
![image](https://github.com/hms-networks/flexy-canary-connector/assets/51409174/178d06a7-fbc0-4dda-ba39-a0db39a3b3a2)
![image](https://github.com/hms-networks/flexy-canary-connector/assets/51409174/1dd647d6-84fd-4556-a83e-be66a02855ce).

Reply:
Updated the comment here, as well as added one for the enclosing `if/else` statement.